src/EPR.Calculator.Frontend/Controllers/CalculationRunDetailsController.cs

```diff
@@ -1,13 +1,125 @@
-﻿using EPR.Calculator.Frontend.Constants;
+﻿using Azure.Core;
+using EPR.Calculator.Frontend.Constants;
+using EPR.Calculator.Frontend.Enums;
+using EPR.Calculator.Frontend.Helpers;
+using EPR.Calculator.Frontend.ViewModels;
+using Humanizer;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.VisualStudio.Web.CodeGenerators.Mvc.Templates.Blazor;
+using Newtonsoft.Json;
+using System.Reflection;
+using static System.Net.Mime.MediaTypeNames;
 
 namespace EPR.Calculator.Frontend.Controllers
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CalculationRunDetailsController"/> class.
+    /// </summary>
     public class CalculationRunDetailsController : Controller
     {
-        public IActionResult Index()
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory clientFactory;
+        private readonly ILogger<CalculationRunDetailsController> logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CalculationRunDetailsController"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration settings.</param>
+        /// <param name="clientFactory">The HTTP client factory.</param>
+        /// <param name="logger">The logger instance.</param>
+        public CalculationRunDetailsController(IConfiguration configuration, IHttpClientFactory clientFactory, ILogger<CalculationRunDetailsController> logger)
+        {
+            this.configuration = configuration;
+            this.clientFactory = clientFactory;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Displays the calculation run details index view.
+        /// </summary>
+        /// <param name="runId">The ID of the calculation run.</param>
+        /// <param name="calcName">The calcName of the calculation run.</param>
+        /// <returns>The calculation run details index view.</returns>
+        public async Task<IActionResult> IndexAsync(int runId, string calcName)
         {
-            return this.View(ViewNames.CalculationRunDetailsIndex);
+            var getCalculationDetailsResponse = await this.GetCalculationDetailsAsync(runId);
```

Comment:
Action methods where it making calls to API I don't see any try catch block. What if any error/exceptions from api?